### ISSUE TITLE
Layerset folding broken with duplex in v0.6

### ISSUE THREAD

**Issue by otsoa** [IfcOpenShell/IfcOpenShell#1368]
![image](https://user-images.githubusercontent.com/99903/110464501-e0a47700-80db-11eb-999e-c13a38754981.png)
The breaking commit seems to be: 75b24c2352 Put maximum length on layerset wall end point intersection
Removing the first condition `d < total_thickness * 3` in following seems to fix it for this model:

```
if (d < total_thickness * 3 && d < mind) {
	GeomAdaptor_Curve GAC(other_axis_curve);
	GeomAdaptor_Surface GAS(kt->first);

	Extrema_ExtCS x(GAC, GAS, getValue(GV_PRECISION), getValue(GV_PRECISION));

	if (x.IsParallel()) {
		body_surface = kt->first;
		layer_body_intersection = kt->second;
		mind = d;
	}
}
```

**Comment by otsoa** [comment]
I just tested with latest v0.6.0 (e62b7e94a4 2021-03-08 16:42) and the issue seems to be there. Maybe it's regression?

**Comment by Francois-Thierry** [comment]
Yes you are right, it is a regression.
Here is the commit that broke it : https://github.com/IfcOpenShell/IfcOpenShell/commit/acb518613bb93b1370941c0fdaf7796d11eec2a5

**Comment by otsoa** [comment]
fold_layers is pretty complex function. I wonder if was possible to write some unit tests for it with f.e. GoogleTest framework. Probably they would save time in the end. I might take a look into that if I have time. @aothms How do you see it? Worth to try?

**Comment by aothms** [comment]
@otsoa definitely! It's schema dependent code so maybe that's a
complicating factor. But I'm definitely in favor of this. Same for the
parsing btw. If you integrate gtest in cmake I'll see if I can contribute
some tests as well.

Sent from a mobile device. Excuse my brevity. Kind regards, Thomas

Op vr 12 mrt. 2021 08:43 schreef otsoa ***@***.***>:

**Comment by aothms** [comment]
Thanks for tracking this down @otsoa  and @Francois-Thierry  It was indeed introduced at https://github.com/IfcOpenShell/IfcOpenShell/commit/acb518613bb93b1370941c0fdaf7796d11eec2a5 which was a fix for #1118 instead of looking for parallel walls the fix should have been to see if the length of the wall can actually accommodate the folds.

**Comment by Francois-Thierry** [comment]
Hi,
I can confirm that the regression is fixed as of 624e296.
Thanks for the quick resolve !
Kind Regards
François

**Comment by otsoa** [comment]
Great job!
